kerchunk/tests/test_grib.py

```diff
@@ -276,3 +279,84 @@ def test_hrrr_sfcf_grib_datatree():
         np.array([0, 3600 * 10**9], dtype="timedelta64[ns]"),
     )
     assert dt.u.attrs == dict(name="U component of wind")
+
+
+def test_parse_grib_idx_invalid_url():
+    with pytest.raises(ValueError):
+        # a random protocol is used
+        parse_grib_idx(
+            "ds://global-forecast-system/gfs.20230928/00/atmos/gfs.t00z.pgrb2.0p25.f001"
+        )
+
+
+def test_parse_grib_idx_no_file():
+    with pytest.raises(FileNotFoundError):
+        # the url is spelled wrong
+        parse_grib_idx(
+            "s3://noaa-hrrr-bdp-pds/hrrr.20220804/conus/hrrr.t01z.wrfsfcf01.grib2",
+            dict(anon=True),
+        )
+
+
+@mock.patch("fsspec.core.url_to_fs")
+def test_parse_grib_idx_duplicate_attrs(mock_url_to_fs):
+    # the "hrrr.t08z.wrfsfcf01.grib2" is not present inside the repo
+    fn = os.path.join(here, "hrrr.t08z.wrfsfcf01.grib2")
+
+    mock_fs = mock.Mock()
+    mock_url_to_fs.return_value = (mock_fs, None)
```

Comment:
Inmstead of doing this mock stuff, you can use memory filesystem ("memory://"), perhaps my taking the `m` fixture from fsspec. The grib file only needs a size, so you can fill it with anything.